nestor_api/lib/config.py

```diff
@@ -1,9 +1,60 @@
 """Configuration library"""
 
+import errno
+import os
 
-def get_app_config(app_name):
-    """Retrieves the configuration of an application"""
+from nestor_api.config.config import Configuration
+from nestor_api.lib.errors import AppConfigurationNotFoundError
+import nestor_api.lib.io as io
+import nestor_api.utils.dict as dict_utils
 
-    # Awaiting for implementation
+APPS_FOLDER = "apps"
 
-    return dict({"app_name": app_name})
+
+def change_environment(environment: str, config_path=Configuration.get_config_path()):
+    """Change the environment (branch) of the configuration"""
+    io.execute("git stash", config_path)
+    io.execute("git fetch origin", config_path)
+    io.execute(f"git checkout {environment}", config_path)
+    io.execute(f"git reset --hard origin/{environment}", config_path)
+
+
+def create_temporary_config_copy():
+    """Copy the configuration into a temporary directory and returns its path"""
+    return io.create_temporary_copy(Configuration.get_config_path(), "config")
+
+
+def get_app_config(app_name: str) -> dict:
+    """Load the configuration of an app"""
+    app_config_path = os.path.join(Configuration.get_config_path(), APPS_FOLDER, f"{app_name}.yaml")
+    if not io.exists(app_config_path):
+        raise AppConfigurationNotFoundError(app_name)
+
+    app_config = io.from_yaml(app_config_path)
```

Comment:
I think we will need to change this eventually because from PyYAML, the parser that uses `safe_load` silently manages bad yaml (like duplicates), I was exploring another library like https://pypi.org/project/ruamel.yaml/ to handle those cases in a more specific manner. Or to override and add constructor to handle cases that PyYAML is not handling properly